docs_user/modules/proc_retrieving-topology-specific-service-configuration.adoc

```diff
@@ -10,186 +10,239 @@ Before you migrate your databases to the {rhos_long} control plane, retrieve the
-"select host, configurations from agents where agents.binary='neutron-sriov-nic-agent';"
-----
-
 . Store the exported variables for future use:
 +
 ----
-$ cat >~/.source_cloud_exported_variables <<EOF
-PULL_OPENSTACK_CONFIGURATION_DATABASES="$PULL_OPENSTACK_CONFIGURATION_DATABASES"
-PULL_OPENSTACK_CONFIGURATION_MYSQLCHECK_NOK="$PULL_OPENSTACK_CONFIGURATION_MYSQLCHECK_NOK"
-PULL_OPENSTACK_CONFIGURATION_NOVADB_MAPPED_CELLS="$PULL_OPENSTACK_CONFIGURATION_NOVADB_MAPPED_CELLS"
-PULL_OPENSTACK_CONFIGURATION_NOVA_COMPUTE_HOSTNAMES="$PULL_OPENSTACK_CONFIGURATION_NOVA_COMPUTE_HOSTNAMES"
-PULL_OPENSTACK_CONFIGURATION_NOVAMANAGE_CELL_MAPPINGS="$PULL_OPENSTACK_CONFIGURATION_NOVAMANAGE_CELL_MAPPINGS"
-SRIOV_AGENTS="$SRIOV_AGENTS"
-EOF
-----
+$ unset SRIOV_AGENTS
+$ unset PULL_OPENSTACK_CONFIGURATION_DATABASES
+$ unset PULL_OPENSTACK_CONFIGURATION_MYSQLCHECK_NOK
+$ unset PULL_OPENSTACK_CONFIGURATION_NOVA_COMPUTE_HOSTNAMES
+$ declare -xA SRIOV_AGENTS <1>
+$ declare -xA PULL_OPENSTACK_CONFIGURATION_DATABASES
+$ declare -xA PULL_OPENSTACK_CONFIGURATION_MYSQLCHECK_NOK
+$ declare -xA PULL_OPENSTACK_CONFIGURATION_NOVA_COMPUTE_HOSTNAMES
+$ for CELL in $(echo $CELLS); do
+>     cat > ~/.source_cloud_exported_variables_$CELL << EOF
+> PULL_OPENSTACK_CONFIGURATION_DATABASES[$CELL]="$(oc run mariadb-client-5-$CELL ${MARIADB_RUN_OVERRIDES} -q --image ${MARIADB_IMAGE} -i --rm --restart=Never -- \
+>     mysql -rsh ${SOURCE_MARIADB_IP[$CELL]} -uroot -p${SOURCE_DB_ROOT_PASSWORD[$CELL]} -e 'SHOW databases;')"
+> PULL_OPENSTACK_CONFIGURATION_MYSQLCHECK_NOK[$CELL]="$(oc run mariadb-client-6-$CELL ${MARIADB_RUN_OVERRIDES} -q --image ${MARIADB_IMAGE} -i --rm --restart=Never -- \
+>     mysqlcheck --all-databases -h ${SOURCE_MARIADB_IP[$CELL]} -u root -p${SOURCE_DB_ROOT_PASSWORD[$CELL]} | grep -v OK)"
+> PULL_OPENSTACK_CONFIGURATION_NOVA_COMPUTE_HOSTNAMES[$CELL]="$(oc run mariadb-client-7-$CELL ${MARIADB_RUN_OVERRIDES} -q --image ${MARIADB_IMAGE} -i --rm --restart=Never -- \
+>     mysql -rsh ${SOURCE_MARIADB_IP[$CELL]} -uroot -p${SOURCE_DB_ROOT_PASSWORD[$CELL]} -e \
+>     "select host from nova.services where services.binary='nova-compute';")"
+> EOF
+> done
+$ cat >> ~/.source_cloud_exported_variables_default << EOF
+> PULL_OPENSTACK_CONFIGURATION_NOVADB_MAPPED_CELLS="$(oc run mariadb-client-2 ${MARIADB_RUN_OVERRIDES} -q --image ${MARIADB_IMAGE} -i --rm --restart=Never -- \
+>     mysql -rsh ${SOURCE_MARIADB_IP['default']} -uroot -p${SOURCE_DB_ROOT_PASSWORD['default']} -e \
+>     'select uuid,name,transport_url,database_connection,disabled from nova_api.cell_mappings;' || echo None)"
+> PULL_OPENSTACK_CONFIGURATION_NOVAMANAGE_CELL_MAPPINGS="$($CONTROLLER1_SSH sudo podman exec -it nova_conductor nova-manage cell_v2 list_cells)"
+> EOF
+$ chmod 0600 ~/.source_cloud_exported_variables*
+----
++
+<1> If `neutron-sriov-nic-agent` agents are running in your {OpenStackShort} deployment, get the configuration to use for the data plane adoption
```

Comment:
```suggestion
<1> If `neutron-sriov-nic-agent` agents are running in your {OpenStackShort} deployment, get the configuration to use for the data plane adoption.
```
Where do you get this configuration?

Reply:
This is probably a formatting change only, I was not introducing this statement in the 1st place, so I cannot answer to that question (@karelyatin perchance? we could address that improvement to SR-IOV adoption guide in follow up)